### ISSUE TITLE
disable aircraft out of ammo behavior.

### ISSUE THREAD

**Issue by UppsAlert** [OpenRA/OpenRA#19427]
The Aircraft out of ammo behavior as it is at the moment isnt micro friendly it lead to create chaos on the battlefield. 
Units try to automatic reload when they are out of ammo for the target. This behavoir displease the "competetive " scene because its almost guaranteed to lose aircraft due to this. 
A idle animation instead of the automatic return to reload would be much better. A move order to the next waypoint, if existent would be great. 
This out of ammo behavior is not implemented for the attack move which makes everything more confusing.

**Comment by Smittytron** [comment]
> This out of ammo behavior is not implemented for the attack move which makes everything more confusing.

Attack move used to have aircraft RTB on Winchester as well. This was changed intentionally but I can't recall why. To me the old RTB behavior makes much more sense than the current behavior as an A-move order is commonly issued behind the targets. This results in empty helicopters flying over enemy airspace and getting shot down.

**Comment by UppsAlert** [comment]
> 
>  A-move order is commonly issued behind the targets. This results in empty helicopters flying over enemy airspace and getting shot down.

Thats another good point units could just stop when they are out of ammo, if not reorderd and not suicide directly into the enemy.

**Comment by penev92** [comment]
Now that would be weird - ordering your units to move only for them to sit still.
IMO the current behaviour is the best compromise as it only requires minimal attention by the player to not fly in unarmed units and lose them. Units ignoring orders, changing their order type by themselves (attack-move -> move) or units deciding to go in the opposite direction (through potentially unknown or enemy airspace) in no way guarantees you are going to lose less of them than the current behaviour.
As a separate point, what happens if the units have multiple ammo pools, different weapons, a weapon with infinite ammo or a self-reloading ammo pool? In half of those cases you're going to get undefined behaviour, while in the other you're going to limit the player from doing what **they** want because **their units** decided otherwise.

**Comment by UppsAlert** [comment]
@penev92 Right now you lose your aircraft because they decide to reload instead of following your orders. Why dont let them follow your orders until you decide its time for reloading? When the ammo pool is empty it could stop/circle over the target or in weapon range. And when you have another waypoint set the aircraft flys to this direction because the job is done.

**Comment by penev92** [comment]
I am not saying the current behaviour is perfect as in "it does everything the player wants with no micro required and is super intuitive and newbie-friendly", bur rather that all the alternatives are worse than or equal to it.

> When the ammo pool is empty it could stop/circle over the target or in weapon range.

Stopping over the target is possibly the worst possible choice as that is likely to be the most dangerous spot.

**Comment by Smittytron** [comment]
The common scenario where this is an issue is when players are controlling a group of several helicopters, and they don't all run out of ammo at the same time. One helicopter at a time suicides over the battle as they run out of ammo.

The old RTB behavior made sense to me because 90% of the time your helicopters are going to be in-between their helipads and the enemy, meaning helis would leave the battle instead of fly over it.

Another alternative I can think of is to have empty helis stay at weapons range facing a target. This would keep the player's desired formation.

**Comment by pchote** [comment]
This topic has already been discussed to death over several issues/PRs and Discord discussions. #17573 explains why AttackMove behaves the way it does now and what the various iterations of C&C-style logic did.

IMO the best solution to the remaining inconsistencies is to make attacks behave the same as attack move does now *and* to make units automatically return to base when they are idle and empty. Forcing players to have to manually order every aircraft back every time it needs to reload isnt an option.

**Comment by UppsAlert** [comment]
> 
> IMO the best solution to the remaining inconsistencies is to make attacks behave the same as attack move does now _and_ to make units automatically return to base when they are idle and empty. Forcing players to have to manually order every aircraft back every time it needs to reload isnt an option.

Maybe I just misinterpret something right now, but a aircraft on a attack move order dont return to base it follows the order until its empty and circle/stops over the end of the targtet line location.
If you direct target a actor and its gone empty or already is empty it return to base. Which makes it special difficult to snipe buildings in enemy bases or atleast cost inefficent.

**Comment by Leo512bit** [comment]
How about airplanes have unlimited ammo but limited fuel like in Rise of Nations?

**Comment by pchote** [comment]
> Maybe I just misinterpret something right now,

That paragraph was suggesting changes that might improve aircraft behaviour, not describing the way things work now.

**Comment by KOYK** [comment]
> This topic has already been discussed to death over several issues/PRs and Discord discussions. #17573 explains why AttackMove behaves the way it does now and what the various iterations of C&C-style logic did.
> 
> IMO the best solution to the remaining inconsistencies is to make attacks behave the same as attack move does now _and_ to make units automatically return to base when they are idle and empty. Forcing players to have to manually order every aircraft back every time it needs to reload isnt an option.

With the danger of getting thumped down by fake accounts (not targeting pchote here, just to be clear). 

And also getting ignored since certain people don't like me, here is my opinion and some possible solutions (that will never be added\fixed) on this one.

The problem with "return to base on idle" is that players use air units as stationary scouts, if fog of war had a cool down timer before it started re-growing back again, it would solve that issue (and many other issues) and return on idle would be easier to be added without penalties on scouting. 
FYI https://github.com/OpenRA/OpenRA/issues/19126 

Also return to base on empty ammo is already in the game, but the only problem with this is that aircrafts block each other on air for ever.
This could be a good opportunity to make aircraft units lose their collusion with other aircraft units, momentarily if they have issued a command to return to base AND if they have empty ammo (to avoid potential abuse).

**To be honest having manually selecting each unit to go reload it will be a nightmare** as pchote mention above with other words. 
And there is already a way to control specific units in the game with control groups, though with that we go back on the mid air collusion problem ive mention above.

**Comment by abcdefg30** [comment]
I agree that the current behaviour is not good, and like the suggestion of https://github.com/OpenRA/OpenRA/issues/16668#issuecomment-500227689 (which is also suggested in the OP here).
